### PR TITLE
updateStatus: Fix error messages containing parameter

### PR DESCRIPTION
**Pull request by lolodomo**
Fixes issue #1014

Also add status CONFIGURATION_ERROR.

Also reduce log level in the thing handlers

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java [hunk 1]
```diff
@@ -141,7 +143,7 @@ public void dispose() {
             serialPort.close();
             serialPort = null;
         }
-        logger.info("Stopped ZWave serial handler");
+        logger.debug("Stopped ZWave serial handler");
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: I think INFO is fine for this? It's inline with the recommendations.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java [hunk 2]
```diff
@@ -199,7 +201,7 @@ private void sendResponse(int response) {
                     logger.trace("Response SENT {}", response);
                 }
             } catch (IOException e) {
-                logger.error("Exception during send", e);
+                logger.debug("Exception during send", e);
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: Probably this should be WARN - this is quite a serious error as commands are not being sent.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java [hunk 3]
```diff
@@ -238,7 +240,7 @@ public void run() {
                             continue;
                         }
                     } catch (IOException e) {
-                        logger.error("Got I/O exception {} during receiving. exiting thread.", e.getLocalizedMessage());
+                        logger.debug("Got I/O exception {} during receiving. exiting thread.", e.getLocalizedMessage());
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java [hunk 4]
```diff
@@ -342,7 +344,7 @@ public void run() {
 
                 }
             } catch (Exception e) {
-                logger.error("Exception during ZWave thread. ", e);
+                logger.debug("Exception during ZWave thread. ", e);
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java [hunk 5]
```diff
@@ -369,7 +371,7 @@ public void sendPacket(SerialMessage serialMessage) {
                 logger.debug("Message SENT");
             }
         } catch (IOException e) {
-            logger.error("Got I/O exception {} during sending. exiting thread.", e.getLocalizedMessage());
+            logger.debug("Got I/O exception {} during sending. exiting thread.", e.getLocalizedMessage());
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 6]
```diff
@@ -130,13 +130,17 @@ public void initialize() {
 
         final BigDecimal cfgNodeId = (BigDecimal) getConfig().get(ZWaveBindingConstants.CONFIGURATION_NODEID);
         if (cfgNodeId == null) {
-            logger.error("NodeID is not set in {}", getThing().getUID());
+            logger.debug("NodeID is not set in {}", getThing().getUID());
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error and the thing will not work.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 7]
```diff
@@ -130,13 +130,17 @@ public void initialize() {
...
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    ZWaveBindingConstants.OFFLINE_NODEID_UNSET);
             return;
         }
 
         nodeId = cfgNodeId.intValue();
         if (nodeId < 1 || nodeId > 232) {
-            logger.error("NodeID ({}) out of range for {}", cfgNodeId, getThing().getUID());
+            logger.debug("NodeID ({}) out of range for {}", cfgNodeId, getThing().getUID());
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 8]
```diff
@@ -264,7 +268,7 @@ void initialiseNode() {
                     try {
                         dataType = DataType.valueOf(bindingType[2]);
                     } catch (IllegalArgumentException e) {
-                        logger.warn("NODE {}: Invalid item type defined {} for {}. Assuming DecimalType.", nodeId,
+                        logger.debug("NODE {}: Invalid item type defined {} for {}. Assuming DecimalType.", nodeId,
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 9]
```diff
@@ -423,7 +427,7 @@ public void run() {
                         controllerHandler.sendData(message);
                     }
                 } catch (Exception e) {
-                    logger.warn("NODE {}: Polling aborted due to exception", nodeId, e);
+                    logger.debug("NODE {}: Polling aborted due to exception", nodeId, e);
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 10]
```diff
@@ -525,7 +529,7 @@ public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
         // Add the listener for ZWave events.
         // This ensures we get called whenever there's an event we might be interested in
         if (bridgeHandler.addEventListener(this) == false) {
-            logger.error("NODE {}: Controller failed to register event handler.", nodeId);
+            logger.debug("NODE {}: Controller failed to register event handler.", nodeId);
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 11]
```diff
@@ -603,7 +607,7 @@ public void handleConfigurationUpdate(Map<String, Object> configurationParameter
             switch (cfg[0]) {
                 case "config":
                     if (cfg.length < 3) {
-                        logger.warn("NODE {}: Configuration invalid {}", nodeId, configurationParameter.getKey());
+                        logger.debug("NODE {}: Configuration invalid {}", nodeId, configurationParameter.getKey());
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 12]
```diff
@@ -951,10 +955,10 @@ public void handleConfigurationUpdate(Map<String, Object> configurationParameter
                                 controllerHandler.sendData(node.encapsulate(userCodeCommandClass.getUserCode(code), 0));
                                 pendingCfg.put(configurationParameter.getKey(), valueObject);
                             } else {
-                                logger.error("Value format error processing user code {}", valueObject);
+                                logger.debug("Value format error processing user code {}", valueObject);
                             }
                         } catch (NumberFormatException e) {
-                            logger.error("Number format exception parsing user code ID '{}'",
+                            logger.debug("Number format exception parsing user code ID '{}'",
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 13]
```diff
@@ -951,10 +955,10 @@ public void handleConfigurationUpdate(Map<String, Object> configurationParameter
                                 controllerHandler.sendData(node.encapsulate(userCodeCommandClass.getUserCode(code), 0));
                                 pendingCfg.put(configurationParameter.getKey(), valueObject);
                             } else {
-                                logger.error("Value format error processing user code {}", valueObject);
+                                logger.debug("Value format error processing user code {}", valueObject);
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 14]
```diff
@@ -1158,7 +1162,7 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
 
         if (cmdChannel.getConverter() == null) {
-            logger.warn("NODE {}: No command converter set for command {} type {}", nodeId, channelUID, dataType);
+            logger.debug("NODE {}: No command converter set for command {} type {}", nodeId, channelUID, dataType);
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 15]
```diff
@@ -1132,7 +1136,7 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         try {
             dataType = DataType.valueOf(command.getClass().getSimpleName());
         } catch (IllegalArgumentException e) {
-            logger.error("NODE {}: Command received with no implementation ({}).", nodeId,
+            logger.debug("NODE {}: Command received with no implementation ({}).", nodeId,
```
Comment (cdjackson) [inline src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Probably this should be WARN - this is quite a serious error.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java [hunk 16]
```diff
@@ -81,7 +81,9 @@ public void initialize() {
         portId = (String) getConfig().get(CONFIGURATION_PORT);
 
         if (portId == null || portId.length() == 0) {
-            logger.error("ZWave port is not set.");
+            logger.debug("ZWave port is not set.");
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: The user will be informed through the thing status.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java (hunk 1 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: I don't think other bindings are logging this information at INFO level.
Your equivalent log in the initialize() method is DEBUG level.
But ok, I resotre it to info.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 6 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: The user will be informed through the thing status.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 7 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: The user will be informed through the thing status.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 8 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java (hunk 5 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java (hunk 4 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java (hunk 3 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java (hunk 2 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 17]
```diff
@@ -1358,7 +1362,7 @@ public void ZWaveIncomingEvent(ZWaveEvent incomingEvent) {
                 }
 
                 if (channel.getConverter() == null) {
-                    logger.warn("NODE {}: No state converter set for channel {}", nodeId, channel.getUID());
+                    logger.debug("NODE {}: No state converter set for channel {}", nodeId, channel.getUID());
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java [hunk 18]
```diff
@@ -1228,7 +1232,7 @@ public void ZWaveIncomingEvent(ZWaveEvent incomingEvent) {
 
                         ZWaveNode node = controllerHandler.getNode(nodeId);
                         if (node == null) {
-                            logger.error("NODE {}: Error getting node for config update", nodeId);
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 9 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 10 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 11 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 12 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 13 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 14 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn.

File: src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java (hunk 15 above)
Comment (lolodomo) [reply src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java]: Changed to warn.

**Comment by cdjackson** [comment]
Thanks.

**Pull request merged**